### ISSUE TITLE
Multiple StoreSubscribers?

### ISSUE THREAD

**Issue by Syrou** [GeoThings/ReKotlin-CounterExample#1]
Since Kotlin/Java does not support multiple implementetion of the same interface more than once
with a specific type (because of type erasure) how do you propose we implement multiple stores in the same Activity?

The issue would be to keep track of several states that I would want to display in several activities.

**Comment by hdrdavies** [comment]
Yes I'm sure this is a very common use case.... Is there any way of achieving it?

**Comment by Syrou** [comment]
My current feeling is that my suggestion/question is pattern breaking, and the solution is to create a ActionCreator that routes the actions you are interested in differently. You can implement a BaseAction that contains shared states like OnError, OnLoading and OnSuccess but if there is just a specific thing that you are interested at a given time, you have to implement it in the specific action setup for a state and then inherit that specific shared action

**Comment by hdrdavies** [comment]
ah I think I might have misread your question, I was wondering how to subscribe to multiple parts of the state tree within the same activity / fragment, but I have solved this. Creating multiple stores is definitely an anti-pattern in redux.
